toplevel/opttopmain.ml

```diff
@@ -20,6 +20,26 @@ let usage =
 
 let preload_objects = ref []
 
+(* Position of the current last expanded argument *)
+let last_expand_pos: int option ref = ref None
+
+let current = ref (!Arg.current)
+
+let argv = ref Sys.argv
+
+(* Test whether the option is part of a responsefile *)
+let is_expanded pos =
+  match !last_expand_pos with
+  | None -> false
+  | Some epos ->  pos <= epos
+
+let expand_position pos len =
+  let pos = match !last_expand_pos with
+    | Some opos when pos < opos ->
+        opos+len (* Already expand option just shift the position *)
+    | _ -> pos + len + 1 in (* New last position *)
```

Comment:
The `+1` is because the current index is at this point the file name. Like all  other `spec` functions `consume_arg` is only called after the function is called.